src/main/java/com/uid2/client/TokenGenerateInput.java

```diff
@@ -0,0 +1,93 @@
+package com.uid2.client;
+
+import com.google.gson.JsonObject;
+
+import java.security.MessageDigest;
+
+public class TokenGenerateInput {
+    public static TokenGenerateInput fromEmail(String email) {
+        return new TokenGenerateInput(IdentityType.Email, email, true);
+    }
+
+    public static TokenGenerateInput fromPhone(String phone) {
+        return new TokenGenerateInput(IdentityType.Phone, phone, true);
+    }
+
+    public TokenGenerateInput withTransparencyAndConsentString(String tcString) {
+        this.transparencyAndConsentString = tcString;
+        return this;
+    }
+
+    TokenGenerateInput doNotHash() {
```

Comment:
Should this be public?

Reply:
we've decided to leave package-private for now; The main reason not to hash was to simplify calling the API, but the SDK takes care of hashing now